<?xml version="1.0"?>
<launch>

    <arg name="barn" default="false" />
    <arg name="barn_dist" default="10.0" />
    <arg name="param_file" default="$(find robust_fast_navigation)/params/planner.yaml" />

    <rosparam file="$(find robust_fast_navigation)/params/planner.yaml" command="load" ns="robust_planner/" />
    <rosparam file="$(find robust_fast_navigation)/params/robo_params.yaml" command="load" ns="robust_planner/" />

    <node name="robust_planner" pkg="robust_fast_navigation" type="planner" output="screen">
    <!--<node name="robust_planner" pkg="cbf_tracking" type="planner" 
           launch-prefix="lldb" output="screen">  -->

           <!-- Trajectory Generation -->
        <!-- <param name="plan_once" value="true" /> -->
        <param name="is_barn" value="$(arg barn)" />
        <param name="barn_goal_dist" value="$(arg barn_dist)" />
        
        <remap from="/planner_goal" to="/move_base_simple/goal" />
        <remap from="/odometry/filtered" to="/gmapping/odometry" />
    </node>

    <rosparam file="$(find robust_fast_navigation)/params/global_costmap_params.yaml" command="load" ns="robust_planner/global_costmap/" />
    <!--<rosparam file="$(find robust_fast_navigation)/params/local_costmap_params.yaml" command="load" ns="robust_planner/local_costmap/" />-->
    
</launch>
